### DISCUSSION TITLE
patterns/state/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#102] [Announcements]
# patterns/state/

Explore the State Pattern, a core component of Java design patterns that enables dynamic behavior change in objects with internal state shifts. Includes real-world examples, applicability, benefits, and detailed code snippets.

https://java-design-patterns.com/patterns/state/

**Comment by igor-zaporozhtsev** [comment]
Every call mammoth.timePasses(); create a new instance. Would it be better for memory if we used a single instance?

**Comment by igor-zaporozhtsev** [comment]
we have link in each State implementation only for printing the name "Mammoth"? :)
```private final Mammoth mammoth;```
